.github/workflows/nightly.yml

```
@@ -0,0 +1,143 @@
+name: Nightly
+on:
+  # FIXME: Change to run on a schedule before merging
+  #schedule:
+  #  - cron: "0 1 * * *"
+  pull_request:
+    branches:
+      - '**'
+jobs:
+  GENERATE_MATRIX:
+    name: Generate Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      branches: ${{ steps.set-matrix.outputs.branches }}
+      asan-matrix: ${{ steps.set-matrix.outputs.asan-matrix }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Get current date for cache
+        id: date
+        run: echo "::set-output name=date::$(date -Iseconds)"
+      - name: Grab the commit mapping
+        uses: actions/cache@v3
+        with:
+          path: branch-commit-cache.json
+          key: nightly-${{ steps.date.outputs.date }}
```

I was wondering if it would be possible to detect whether the "generate Matrix" job was manually reattempted and then forcing a fresh build. While doing so I came across [`github.run_id`](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) which allows you to simplify this a bit:

```suggestion
          key: nightly-${{ github.run_id }}-${{ github.run_attempt }}
```
(You can remove the `date` step then).